src/main/java/duke/parser/CommandParser.java

```diff
@@ -0,0 +1,142 @@
+        return new AddEventOperation(description, parsedTime, list);
+    }
+
+    private DeleteOperation createDeleteOp(String[] commands, TaskList list) throws DukeParseException {
+        if (CommandType.DELETE.isValidLength(commands.length)) {
+            throw new DukeParseException("Ensure a number is passed after a delete command.");
+        }
+        if (!Utils.hasInteger(commands, 1)) {
+            throw new DukeParseException("Ensure a number is passed after a delete command.");
+        }
+        int index = Integer.parseInt(commands[1]);
+
+        return new DeleteOperation(list, index);
+    }
+
+    private FindOperation createFindOp(String[] commands, TaskList list) throws DukeParseException {
+        if (CommandType.FIND.isValidLength(commands.length)) {
+            throw new DukeParseException(
+                    "Ensure a keyword is entered so that I can perform a search with it.");
+        }
+        String searchWord = Utils.concatenate(commands, 1, commands.length);
+        return new FindOperation(list, searchWord);
+    }
+
+    /**
+     * Parses the String given into an <code>Operation</code>.
+     *
+     * @param commandString the <code>String</code> that has been input by the user into Duke.
+     * @param list the <code>TaskList</code> to be operated on.
+     * @param taskStorage the <code>TaskStorage</code> to be operated on,
+     *                    if the <code>Operation</code> requires a save of the <code>TaskList</code>.
+     * @return the parsed <code>Operation</code>.
+     * @throws DukeParseException if the command cannot be recognised or is erroneous.
+     */
+    public Operation parse(String commandString, TaskList list, TaskStorage taskStorage)
+            throws DukeParseException {
+        String[] commands = commandString.split(" ");
+        Function<CommandType, Boolean> isCommand = commandType ->
+                commandType.getCommand().equals(commands[0]);
+
+        if (isCommand.apply(CommandType.BYE)) {
+            return createExitOp(taskStorage, list);
+        } else if (isCommand.apply(CommandType.LIST)) {
+            return createListOp(list);
+        } else if (isCommand.apply(CommandType.DONE)) {
+            return createDoneOp(commands, list);
+        } else if (isCommand.apply(CommandType.TODO)) {
+            return createTodoOp(commands, list);
+        } else if (isCommand.apply(CommandType.DEADLINE)) {
+            return createDeadlineOp(commands, list);
+        } else if (isCommand.apply(CommandType.EVENT)) {
+            return createEventOp(commands, list);
+        } else if (isCommand.apply(CommandType.DELETE)) {
+            return createDeleteOp(commands, list);
+        } else if (isCommand.apply(CommandType.FIND)) {
+            return createFindOp(commands, list);
+        } else {
+            throw new DukeParseException("This command is not recognised unfortunately.");
+        }
+    }
```

Comment:
I like how you have abstracted out all the parse logic into individual methods, thereby making your parse method easily readable!